File: Java/1451-1500/1470. Shuffle the Array.java
// https://leetcode.com/problems/shuffle-the-array/

class Solution {
    public int[] shuffle(int[] nums, int n) {
        int startIndex = 0, midIndex =n, index = 0;
        int[] result = new int[nums.length];

        while(index < nums.length){
            result[index++] = nums[startIndex++];
            result[index++] = nums[midIndex++];
        }

        return result;
    }
}

File: Javascript/1451-1500/1470. Shuffle the Array.js
// https://leetcode.com/problems/shuffle-the-array/

var shuffle = function(nums, n) {
    const result = [];

    for(let index=0; index<n; index++){
        result.push(nums[index]);
        result.push(nums[n+index]);
    }

    return result;
};
